### ISSUE TITLE
Future Chrome versions have already broken docking

### ISSUE THREAD

**Issue by jcrowthe** [old-hangouts/old-hangouts-chrome-extension#1]
Current version of Chrome on Mac is 53.0.2785.143. 
Current version of Chrome Canary on Mac is 56.0.2897.0 (this will change with nightly build tomorrow)

Using Chrome Canary (https://www.google.com/chrome/browser/canary.html) we can see that the coming versions of Chrome break the docking of this extension, turning it into a free-floating window that does not snap to anything.

I noticed this about 1-2 months ago, before the mandatory "upgrade" happened this week. Due to the rollout nature of Chrome found here https://www.chromium.org/getting-involved/dev-channel, this means that this wonderful build of Hangouts will lose its docking capabilities in 3-6 weeks when these changes are pushed into Chrome Stable. How sad!

We should look into finding the code that docks the window and see if we can find why the update to Chrome is causing the malfunction. We have a few weeks to do it!

**Comment by sonburn** [comment]
I lost docking ability with the last Mac update sadly, I am running Version 54.0.2840.71 (64-bit).

**Comment by alvaro-lau** [comment]
Indeed, Even though I installed this hangout version, still floating windows :( 
I'm using version 55.0.2883.21 beta-m (64-bit).

**Comment by liamfmvt** [comment]
I'm on Version 54.0.2840.59 beta (64-bit) (mac) and same issue - floating windows.  Subscribed in hopes of a fix.

**Comment by pranav-prakash** [comment]
Confirmed. Chrome 54 [removes](https://bugs.chromium.org/p/chromium/issues/detail?id=467808) the panels features that docking depends on. 

> Panels on non-chromeos were an experimental feature and only supported on Dev channel, or on other channels behind a flag.
> 
> They are costly to maintain and we were never happy with their stability and functionality on Windows, Mac and Linux. As a result we've removed them.

The only workaround at this point is to either use chrome 53 with updates disabled, or to run a second headless instance of chrome 53 where you run hangouts.

**Comment by fancellu** [comment]
Ditto. I had to install chromium 53.

**Comment by rainabba** [comment]
Every day Google smells a bit more fruity; telling us what we'll like instead of listening to us :(

**Comment by pranav-prakash** [comment]
Note: If you revert to chrome 53 on osx you have to change permissions on GoogleUpdater to break the auto update mechanism.

**Comment by scytheria1001** [comment]
Got it working by going back to Chrome 53. Changing permissions on GoogleUpdater didn't work for me, Chrome just changed permissions back when I opened it and updated anyway. I ended up deleting GoogleSoftwareUpdateDaemon and GoogleSoftwareUpdateAgent, which worked.

**Comment by patriciamayo** [comment]
Hi all!
Thank you very much for your input. We thought that the update was just happening from the Hangouts team side but as you all mention here, it is due to Chrome updating to version 54, where it won't longer give support for panels (the feature in charge of docking the floating windows).

**Solution: [install Chromium version 52.0.2743.0](https://github.com/old-hangouts/Chromium-52.0.2743.0).** 

This way Chromium 52 (open source project of Chrome in an older version that still supports panels) and Chrome (with most recent updates) can be running at the same time. **We highly advice not to stop updating your Chrome browser**

We have opened a new [repository and tutorial](https://github.com/old-hangouts/Chromium-52.0.2743.0) for Mac and Windows. The README file in this repository has also been updated accordingly.

We are still leaving the issue open as the Chromium solution for Linux is still pending.

**Comment by jcrowthe** [comment]
@patriciamayo May I ask why you selected Chrome 52 instead of Chrome 53? Panels work as expected in Chrome 53.

**Comment by sandoche** [comment]
@jcrowthe I tested different versions of Chromium 53.\* (in windows and mac) and for some reason the dock panel didn't work. That's why we decided to use Chromium 52.*

But it's true that the dock is still working in my Chrome current version (53.*) 

Feel free to give a try maybe I did something wrong
- http://commondatastorage.googleapis.com/chromium-browser-snapshots/index.html?prefix=Win/403380/ 
- http://commondatastorage.googleapis.com/chromium-browser-snapshots/index.html?prefix=Mac/403380/

**Comment by ajayyy** [comment]
You can get close to the old experience by using an always on top program to force them to always be on top.

**Comment by sonburn** [comment]
It wasn't just that the windows were always on top, the windows were "docked" off the bottom the screen and expandable/collapsable.

**Comment by ajayyy** [comment]
I know, I wish I could have that too, but installing and running another version of chrome is just too much. For people who do not want to do that should download this extention and an always on top program.

**Comment by sandoche** [comment]
@ajayyy that's a good alternative for whom doesn't want to have two browsers. Do you have any 'always on top program' to suggest ? I tried one in Mac OSX it didn't work :/

**Comment by ajayyy** [comment]
@sandoche I used a simple Auto Hotkey script `^SPACE::  Winset, Alwaysontop, , A`. There is also a premade version. But this is only for Windows. I'm not sure if there is a thing like this for OSX or Linux.

**Comment by pranav-prakash** [comment]
If anyone here who still has the extension working, please make a backup of chrome cache and upload it.

See :https://github.com/old-hangouts/old-hangouts-chrome-extension/issues/3
